addons/binding/org.openhab.binding.lightify/src/main/java/org/openhab/binding/lightify/handler/GatewayHandler.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Hashtable;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_ADDRESS;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_GATEWAY;
+
+/**
+ * The {@link org.eclipse.smarthome.core.thing.binding.BridgeHandler} implementation to handle commands
+ * and status of the OSRAM Lightify gateway device. This handler uses a native implementation of the
+ * Lightify proprietary protocol to communicate with the Lightify gateway, therefore no internet connection
+ * is necessary to utilize the OSRAM public API.
+ *
+ * @author Christoph Engelbert (@noctarius2k) - Initial contribution
+ */
+public class GatewayHandler extends BaseBridgeHandler {
+
+    /**
+     * Supported {@link ThingTypeUID}s for this handler
+     */
+    public static final Set<ThingTypeUID> SUPPORTED_TYPES = Collections.singleton(THING_TYPE_LIGHTIFY_GATEWAY);
+
+    private LightifyDeviceDiscoveryService discoveryService;
+    private ServiceRegistration<?> serviceRegistration;
+
+    private String address;
+    private LightifyLink lightifyLink;
+    private ScheduledFuture<?> futureConnect;
+
+    public GatewayHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+    }
+
+    @Override
+    public void initialize() {
+        address = getConfig().get(PROPERTY_ADDRESS).toString();
+        scheduleConnect(true);
+        serviceRegistration = registerDeviceDiscoveryService();
+    }
+
+    @Override
+    public void dispose() {
+        futureConnect.cancel(true);
+        if (lightifyLink != null) {
+            lightifyLink.disconnect();
+        }
+        if (serviceRegistration != null) {
+            if (discoveryService != null) {
+                discoveryService.deactivate();
+                discoveryService = null;
+            }
+            serviceRegistration.unregister();
+            serviceRegistration = null;
+        }
+    }
+
+    /**
+     * Provides access to the {@link LightifyLink} used to communicate with the Lightify gateway
+     *
+     * @return the current communication link
+     */
+    public LightifyLink getLightifyLink() {
+        return lightifyLink;
+    }
+
+    private ServiceRegistration<?> registerDeviceDiscoveryService() {
+        discoveryService = new LightifyDeviceDiscoveryService(this);
+        discoveryService.activate();
+        return registerService(DiscoveryService.class, discoveryService);
+    }
+
+    private void scheduleConnect(boolean immediate) {
+        final long delay = immediate ? 0 : 10;
+        if (futureConnect != null) {
+            futureConnect.cancel(true);
+            futureConnect = null;
+        }
+        futureConnect = scheduler.schedule(() -> {
+            lightifyLink = new LightifyLink(address, scheduler);
```

Comment:
That's a clear sign that code reviews should be even stricter, so that no bad examples make it into the repo as they sooner or later will by copy & pasted ;-)